src/wrapper/wrapper.c

```diff
@@ -715,9 +719,13 @@ real_call:; // semicolon here to avoid the error
     }
   #endif // DYAD_SYNC_DIR
 
-    rc = func_ptr (fd);
+    int wronly = is_wronly (fd);
 
-    if (to_sync) {
+    if (wronly == -1)
+        DPRINTF ("Failed to check the mode of the file with fcntl: %s\n", strerror (errno));
```

Comment:
There needs to be {} because DPRINTF is macro that may be undefined.

Reply:
Thanks for the catch. Fixed.